MN/Algoritmi/Metode-Iterative/Gauss-Seidel/GaussSeidel.m

```diff
@@ -0,0 +1,34 @@
+function [x step]= GaussSeidel ( A, b, x0, tol, max_iter)
+    %use this algorithm only if 
+    %max(abs(eig(Ggs))) is < 1
```

Comment:
Spatiu dupa '%'. In loc sa scrii in comentarii o chestie matematica pe care probabil n-o intelege nimeni,poti scrie:
"use this algorithm only if it converges".
Apoi chiar faci verificarile. Ai putea face asa:
```matlab
% sanity checks
N = tril(A);
P = N - A;
Ggs = inv(N) * P;
% calculate the spectral radius of Ggs
sr = max(abs(eig(Ggs)))
% check if the algorithm converges
if abs(sr - 1) >= eps
    x = NaN
    step = -1
    return
endif
```